File: SpringBootProject/src/main/java/fr/iut/sciencequest/sae/dto/partie/PartieDTO.java
package fr.iut.sciencequest.sae.dto.partie;

import fr.iut.sciencequest.sae.dto.difficulte.DifficulteSimpleDTO;
import fr.iut.sciencequest.sae.dto.jeu.JeuDTO;
import fr.iut.sciencequest.sae.dto.joueur.JoueurSimpleDTO;
import fr.iut.sciencequest.sae.dto.thematique.ThematiqueSimpleDTO;
import fr.iut.sciencequest.sae.entities.Status;
import fr.iut.sciencequest.sae.entities.Thematique;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.springframework.hateoas.RepresentationModel;

import java.util.List;
import java.util.Set;


@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class PartieDTO extends RepresentationModel<PartieDTO> {
    @NotNull
    private Integer id;
    @NotEmpty
    private String codeInvitation;
    @NotEmpty
    private List<JoueurSimpleDTO> joueurs;
    @NotEmpty
    private List<ThematiqueSimpleDTO> thematiques;
    @NotNull
    private DifficulteSimpleDTO difficulte;
    @NotNull
    private Status status;
}

File: android/app/src/main/java/fr/iut/sciencequest/model/dto/partie/PartieDTO.kt
package fr.iut.sciencequest.model.dto.partie

import fr.iut.sciencequest.model.dto.JeuDTO
import fr.iut.sciencequest.model.dto.ThematiqueDTO
import fr.iut.sciencequest.model.dto.joueur.JoueurSimpleDTO

class PartieDTO (
    val id: Int,
    val codeInvitation: String,
    val joueurs: List<JoueurSimpleDTO>,
    val jeu: JeuDTO,
    val thematiques: List<ThematiqueDTO>
)